Marlin/Marlin_main.cpp

```diff
@@ -3576,10 +3576,28 @@ inline void gcode_M115() {
  * M117: Set LCD Status Message
  */
 inline void gcode_M117() {
-  char* codepos = strchr_pointer + 5;
+  millis_t codenum = 0;
+  char* codepos;
+  if (code_seen('S')) {
```

Comment:
This is what `code_has_value()` is designed for. It allows the message to contain an "S" but will treat "`S0`" as an intentional value. (Of course, since zero is meaningless the full test should be `if (code_seen('S') && code_has_value() && code_value_long()) { ... }`.

And as long as this is being extended to do the same thing as `G4`, we might as well also take the same parameters, including `P`, following the same pattern.